src/Raven.Server/Web/Studio/DataDirectoryInfo.cs

```diff
@@ -0,0 +1,157 @@
+        {
+            var httpClient = RequestExecutor.CreateHttpClient(_serverStore.Server.Certificate.Certificate, DocumentConventions.Default);
+            var tasks = new List<Task<SingleNodeDataDirectoryResult>>();
+
+            foreach (var nodeTag in nodes)
+            {
+                _serverStore.ServerShutdown.ThrowIfCancellationRequested();
+
+                if (nodeTag.Equals(_serverStore.NodeTag, StringComparison.OrdinalIgnoreCase))
+                    continue;
+
+                var serverUrl = clusterTopology.GetUrlFromTag(nodeTag);
+                if (serverUrl == null)
+                    continue;
+
+                tasks.Add(Task.Run(async () =>
+                {
+                    var url = $"{serverUrl}/{urlPath}";
+                    var singleNodeResult = await GetSingleNodeDataDirectoryInfo(url, httpClient);
+                    return singleNodeResult;
+                }, _serverStore.ServerShutdown));
+            }
+
+            await Task.WhenAll(tasks);
+
+            foreach (var task in tasks)
+            {
+                if (task.IsCompletedSuccessfully == false)
+                    continue;
+
+                var singleNodeResult = await task;
+                if (singleNodeResult == null)
+                    continue;
+
+                dataDirectoryResult.List.Add(singleNodeResult);
+            }
+        }
+
+        private async Task<SingleNodeDataDirectoryResult> GetSingleNodeDataDirectoryInfo(string url, HttpClient httpClient)
+        {
+            using (var cts = new CancellationTokenSource(7 * 1000))
+            {
+                try
+                {
+                    var request = new HttpRequestMessage(HttpMethod.Get, url);
+                    var response = await httpClient.SendAsync(request, cts.Token);
+                    if (response.IsSuccessStatusCode == false)
+                        return null;
+
+                    using (var responseStream = await response.Content.ReadAsStreamAsync())
+                    {
+                        var singleNodeDataBlittable = await _context.ReadForMemoryAsync(responseStream, "studio-tasks-full-data-directory");
+                        return JsonDeserializationServer.SingleNodeDataDirectoryResult(singleNodeDataBlittable);
+                    }
+                }
+                catch (OperationCanceledException)
+                {
+                    return null;
+                }
+                catch (Exception e)
+                {
+                    if (Logger.IsInfoEnabled)
+                        Logger.Info($"Failed to get directory info result from: {url}", e);
+
+                    return null;
+                }
+            }
+        }
+    }
+
+    public class SingleNodeDataDirectoryResult : IDynamicJson
+    {
+        public string NodeTag { get; set; }
+
+        public string FullPath { get; set; }
+
+        public string TotalFreeSpaceHumane { get; set; }
+
```

Comment:
Better to also include the value as a number

Reply:
I do not see those?

Reply:
pushed my changes